### PR TITLE
Cleanup ASCIIEncoding tests

### PR DESCRIPTION
**Pull request by hughbe**
Use common helper that contains more tests than before

/cc @stephentoub @tarekgh

### CODE REVIEW

File: src/System.Text.Encoding/tests/NegativeEncodingTests.cs [hunk 1]
```diff
@@ -21,7 +22,14 @@ public static IEnumerable<object[]> Encodings_TestData()
...
             // Chars is null
-            Assert.Throws<ArgumentNullException>("s", () => encoding.GetByteCount((string)null));
+            if (encoding is ASCIIEncoding)
+            {
+                Assert.Throws<ArgumentNullException>("chars", () => encoding.GetByteCount((string)null));
+            }
+            else
+            {
+                Assert.Throws<ArgumentNullException>("s", () => encoding.GetByteCount((string)null));
+            }
```
Comment (stephentoub) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: Nit:
```C#
Assert.Throws<ArgumentNullException>(encoding is ASCIIEncoding? "chars" : "s", () => encoding.GetByteCount((string)null));
```

File: src/System.Text.Encoding/tests/NegativeEncodingTests.cs [hunk 2]
```diff
@@ -168,10 +181,18 @@ public static void GetString_Invalid(Encoding encoding)
...
+            if (encoding is ASCIIEncoding)
+            {
+                Assert.Throws<ArgumentOutOfRangeException>("byteIndex", () => encoding.GetString(new byte[1], -1, 0));
+                Assert.Throws<ArgumentOutOfRangeException>("byteCount", () => encoding.GetString(new byte[1], 0, -1));
+            }
+            else
+            {
+                Assert.Throws<ArgumentOutOfRangeException>("index", () => encoding.GetString(new byte[1], -1, 0));
+                Assert.Throws<ArgumentOutOfRangeException>("count", () => encoding.GetString(new byte[1], 0, -1));
+            }
```
Comment (stephentoub) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: As noted in another comment, we shouldn't duplicate the entire statements, and should instead simply make the argument name conditional.

**Comment by stephentoub** [comment]
A few nits, otherwise LGTM.

File: src/System.Text.Encoding/tests/NegativeEncodingTests.cs (hunk 2 above)
Comment (hughbe) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: fixed in latest commit

File: src/System.Text.Encoding/tests/NegativeEncodingTests.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: fixed in latest commit

**Comment by tarekgh** [comment]
LGTM

**Pull request merged**